### DISCUSSION TITLE
flow and node_dim

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2144] [Q&A] [answered]
What exactly do the parameters `flow` and `node_dim` for `MessagePassing` determine?

**Comment by rusty1s** [comment, marked as answer]
* `flow` dicates the flow of message passing, which corresponds to the way nodes send messages to their neighbors. In case `flow="source_to_target"`, nodes will sent messages along their edges (a source node sends a message and a target node receives a message). `flow="target_to_source"` reverses this computation flow. This might be useful if you want to distinguish messages in an directed graph (where you also want to send messages in the reversed graph).

* `node_dim` denotes the dimension in which you want to aggregate node features. This is set to `node_dim=-2` by default, but you may want to customize it in case your node features do not have shape `[num_nodes, num_features]`. For example, you may think of temporal node features of shape `[num_nodes, seq_length, num_features]`, where you want to aggregate node features in the first dimension. Here, you need to set `node_dim` to `0` or `-3`.
